server/src/test/java/org/elasticsearch/indices/settings/UpdateSettingsIT.java

```diff
@@ -126,6 +128,16 @@ public void onIndexModule(IndexModule indexModule) {
         }
     }
 
+    /**
+     * Needed by {@link UpdateSettingsIT#testEngineGCDeletesSetting()}
+     */
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder().put(super.nodeSettings(nodeOrdinal))
+            .put("thread_pool.estimated_time_interval", TimeValue.timeValueMillis(1))
```

Comment:
This effectively makes this thread busy spin for this entire test suite. On a machine with a low core count or that is otherwise busy (think how we fork tests across JVMs) it might be too much?

Reply:
We can increase it a bit I guess. The reason of changing the setting here, is to avoid adding a public test-only method to force update of cached time, to avoid waiting too much in the while loop.

Reply:
Maybe we can change the logic of the cache - if this is 0, it disables caching.

Reply:
@bleskes Something like this: https://github.com/elastic/elasticsearch/pull/38942/commits/7fed99ae0f5f0ba1bda80246718e8f060e3f2e6e ?